### DISCUSSION TITLE
How to customize the directory structure of AirSim to the UE project ？

### DISCUSSION THREAD

**Discussion by LittleTimeWxk** [microsoft/AirSim#3916] [Support Q&A] [answered]
I have created a UE project. I don’t want to use the official tutorial to apply AirSim to the UE project, I want to separate the Content folder and Source folder under the airsim folder into my UE project. So I don't know how to build and using AirSim plugin.

I have tried to using official tutorial to clean and generate AirSim that the folders has separated, but it doesn't work. When I open the UE editor and there is no airsim plugin generated.

**Comment by zimmy87** [comment, marked as answer]
I'm a little bit confused by what you mean by "there is no airsim plugin generated"; do you mean you no longer see the plugin listed in the Plugin UI in the Editor? If you remove the content and source of a plugin and place them into your project's main directory structure, it wouldn't surprise me if unreal no longer detects the plugin as present. You may need to make changes to your project's Build.cs to properly load dependencies and configuration needed for AirSim. I would compare and contrast with AirSim's Build.cs to see if there's anything you need to change in your project's Build.cs.

**Reply by LittleTimeWxk** [reply]
Thanks your reply, that works for me. The main reason is that I lack knowledge of UE usage.
By the way, I didn't describe it clearly and "there is no airsim plugin generated" means that the source code content of AirSim cannot be seen in the source code section under the directory structure of the UE editor interface.
